File: Parte1PreProcesadoDeDatos/missing_data.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.impute import SimpleImputer


# Importar el dataSet
# Para mostrar el dataFrame del dataset debemos debuguear la variable
dataset = pd.read_csv("Data.csv")
# localizar elementos por posicion
x = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 3].values

# tratamiento de los NAs
# Reemplazamiento de los valores desconocidos (0=columna 1=fila)
imputer = SimpleImputer(missing_values=np.nan, strategy="mean")
imputer.fit(x[:, 1:3])
x[:, 1:3] = imputer.transform(x[:, 1:3])

File: Parte1PreProcesadoDeDatos/missing_data.R
#Plantilla para el pre procesado de datos faltantes
#Importar el dataset
#Getwd()para saber en que directorio estamos
setwd("Parte1PreProcesadoDeDatos/")
dataset = read.csv('Parte1PreProcesadoDeDatos/Data.csv')
View(dataset)

#Tratamiento de los valores NAs
dataset$Age = ifelse(is.na(dataset$Age),
  ave(dataset$Age,FUN = function (x)mean(x,na.rm=TRUE)),
  dataset$Age)

dataset$Salary = ifelse(is.na(dataset$Salary),
  ave(dataset$Salary,FUN = function (x)mean(x,na.rm=TRUE)),
  dataset$Salary)